### PR TITLE
haskell: Add a maintainer script for opening a new haskell-updates PR

### PR DESCRIPTION
**Pull request by cdepillabout**
###### Motivation for this change

This is a maintainer script for that automates merging the currently open haskell-updates PR, and creating the new PR.

I used this script for merging https://github.com/NixOS/nixpkgs/pull/137340 and opening https://github.com/NixOS/nixpkgs/pull/138316.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cdepillabout** [comment]
This could be merged in as-is, but I also want to add documentation to HACKING.md about how to use it and where it fits in to our workflow.

File: pkgs/development/haskell-modules/HACKING.md [hunk 1]
```diff
@@ -180,24 +153,6 @@ following will happen:
...
-rebuilding Haskell packages.
-
-In order to make sure that there are a minimal number of rebuilds after merging
-`haskell-updates` into `master`, `master` should occasionally be merged into
-the `haskell-updates` branch.
-
-This is especially important after `staging-next` is merged into `master`,
-since there is a high chance that this will cause all the Haskell packages to
-rebuild.
-
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/HACKING.md:200]: We have a GitHub Action that merges `master` into `haskell-updates` automatically now, so I thought we no longer needed this section.

**Comment by cdepillabout** [comment]
I've added an explanation of how to use this script to `HACKING.md`, so this PR should now be ready to be reviewed.

**Review by maralorn — APPROVED** [review]
I obviously haven‘t tested this, but it looks great. And should be merged, imo.

Thanks for improving the documentation so thoroughly.

Maybe one thing missing: Maybe we want the call to update the distributed versions on hackage in this script? (No merge blocker.)

**Comment by cdepillabout** [comment]
@maralorn Thanks for reviewing this.  I'll go ahead and merge it.

> Maybe we want the call to update the distributed versions on hackage in this script?

Yeah, that's a good idea.  I'll add that next time I run the script.

**Pull request merged**